lib/modules/console_listener/index.js

```diff
@@ -0,0 +1,79 @@
+const utils = require('../../utils/utils.js');
+
+class ConsoleListener {
+    constructor(embark, options) {
+        this.logger = embark.logger;
+        this.ipc = options.ipc;  
+        this.addressToContract = [];
+        this.contractsConfig = embark.config.contractsConfig;
+        this.listenForLogRequests();
+    }
+
+    _updateContractList(){
+        Object.keys(this.contractsConfig.contracts).forEach(contractName => {
+            let contract = this.contractsConfig.contracts[contractName];
+            let address = contract.deployedAddress.toLowerCase();
+
+            if(!this.addressToContract[address]){
+                let funcSignatures = {};
+                contract.abiDefinition
+                    .filter(func => func.type == "function")
+                    .map(func => {
+                                return {
+                                    name: func.name + 
+                                            '(' + 
+                                            (func.inputs ? func.inputs.map(input => input.type).join(',') : '') +
+                                            ')',
+                                    abi: func,
+                                    functionName: func.name
+                                };
+                        })
+                    .forEach(func => {
+                        funcSignatures[utils.sha3(func.name).substring(0, 10)] = func;
+                    });
+
+                this.addressToContract[address] = {
+                    name: contractName,
+                    functions: funcSignatures
+                };
+            }
+        });
+    }
+
+    listenForLogRequests(){
+        if(this.ipc.ipcRole === 'server'){
+            this.ipc.on('log', (request) => {
+                if(request.type == 'contract-log'){
+
+                    let {address, data, transactionHash, blockNumber, gasUsed, status} = request;
+                    if(!this.addressToContract[address]){
+                        this._updateContractList();
+                    }
+
+                    let name = this.addressToContract[address].name;
+                    let func = this.addressToContract[address].functions[data.substring(0, 10)]; 
+                    let functionName = func.functionName;
```

Comment:
All those can be `const`s instead of `let`s.

Reply:
Basically, whenever a variable's value doesn't change, you should use `const`, so it is easier to understand that it won't change